### PR TITLE
mat64: remove gocheck dependency

### PR DESCRIPTION
**Pull request by kortschak**
The text of many of the test errors should be improved, but that can be a subsequent PR which should go over the tests and clean up some of the older style that persisted from biogo.matrix.

@btracey Please take a look.

### CODE REVIEW

**Comment by jonlawlor** [comment]
Nice.

File: mat64/dense_test.go [hunk 1]
```diff
@@ -881,29 +1017,30 @@ func (s *S) TestCloneT(c *check.C) {
...
-		r.Clone(a.T())
-		c.Check(Equal(&r, t), check.Equals, true, check.Commentf("Test %d: %v transpose = %v", i, test.a, test.t))
-
-		rr.Clone(r.T())
-		c.Check(Equal(&rr, a), check.Equals, true, check.Commentf("Test %d: %v transpose = %v", i, test.a, test.t))
-
-		zero(r.mat.Data)
-		r.Clone(a.T())
-		c.Check(Equal(&r, t), check.Equals, true, check.Commentf("Test %d: %v transpose = %v", i, test.a, test.t))
+		for j := 0; j < 2; j++ {
```
Comment (btracey) [inline mat64/dense_test.go]: This is nice.

File: mat64/dense_test.go [hunk 2]
```diff
@@ -1234,16 +1390,17 @@ func (s *S) TestOuter(c *check.C) {
...
-		c.Check(Equal(&m, want), check.Equals, true, check.Commentf("Test %v. Want %v, Got %v", i, want, m))
-		// Check with the same matrix
-		m.Outer(NewVector(len(test.x), test.x), NewVector(len(test.y), test.y))
-		c.Check(Equal(&m, want), check.Equals, true, check.Commentf("Test %v. Want %v, Got %v", i, want, m))
+		for j := 0; j < 2; j++ {
+			// Check with a new matrix - and then again.
+			m.Outer(NewVector(len(test.x), test.x), NewVector(len(test.y), test.y))
+			if !Equal(&m, want) {
+				t.Errorf("unexpected result for Outer test %d iteration %d: got: %+v want: %+v", i, j, m, want)
+			}
```
Comment (btracey) [inline mat64/dense_test.go]: zero here?

**Comment by btracey** [comment]
LGTM with minor comment. Thank you.

We should consider using floats.Same / floats.Equal for []float64 instead of reflect where appropriate.

File: mat64/dense_test.go (hunk 2 above)
Comment (kortschak) [inline mat64/dense_test.go]: That would be a logic change, and I didn't write the original test, so I don't want to do that. If you think there should have been a zero between the two calls (at L1240 in the original) then you should send a PR.

**Comment by kortschak** [comment]
>We should consider using floats.Same / floats.Equal for []float64 instead of reflect where appropriate.

Agreed. There are many things to clean up, but now it can happen at a more reasonable pace.

**Comment by btracey** [comment]
Yep, totally agreed. Just wanted to mention.

**Pull request merged**